lib/std/os.zig

```diff
@@ -1140,6 +1140,27 @@ pub fn freeNullDelimitedEnvMap(allocator: *mem.Allocator, envp_buf: []?[*:0]u8)
     allocator.free(envp_buf);
 }
 
+pub const LockCmd = enum {
+    GetLock,
+    SetLock,
+    SetLockBlocking,
+};
+
+/// Attempts to get lock the file, blocking if the file is locked.
+pub fn fcntlFlock(fd: fd_t, lock_cmd: LockCmd, flock_p: *const Flock) OpenError!void {
```

Comment:
I think only a generic wrapper for `fcntl` goes in `os.zig` and then have a function that calls `std.os.fcntl` with the appropriate `F_{GET,SET}LK` flags in `fs.zig`.